pkg/cli/admin/release/extract.go

```diff
@@ -377,18 +377,25 @@ func (o *ExtractOptions) Run() error {
 						continue
 					}
 				}
-				credRequestManifests = append(credRequestManifests, m)
+				o.Manifests = append(o.Manifests, m)
 			}
-			if len(credRequestManifests) == 0 {
-				return true, nil
-			}
-			for _, m := range credRequestManifests {
+			for _, m := range o.Manifests {
 				yamlBytes, err := yaml.JSONToYAML(m.Raw)
 				if err != nil {
 					return false, errors.Wrapf(err, "error serializing manifest in %s", hdr.Name)
 				}
-				fmt.Fprintf(o.Out, "---\n")
-				o.Out.Write(yamlBytes)
+				out := o.Out
+				if len(o.Directory) > 0 {
+					out, err = os.Create(path.Join(o.Directory, hdr.Name))
```

Comment:
The creation of the file should happen outside the loop of manifests. Otherwise for files that contain multiple manifests, you'll create the file multiple times, leaving it only with the last manifest.
Also s/`path.Join`/`filepath.Join`/ so windows clients work.
After creating a file outside of the loop, you also need to add  a `defer out.Close()`